enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,592 @@
+may inadvertently use both the public and private subnets, unknowingly introducing
+a security risk (see related [RFE-2816](https://issues.redhat.com/browse/RFE-2816) and
+[OCPBUGS-17432](https://issues.redhat.com/browse/OCPBUGS-17432)).
+
+As a workaround to this type of issue, cluster admins may want to specify the private
+subnet list to ensure that the default IngressController exclusively uses the private
+subnets.
+
+See [Setting IngressController Subnets during Installation Workflow](#setting-ingresscontroller-subnets-during-installation-workflow)
+for the workflow for this user story.
+
+### Goals
+
+- Introduce a new field in the IngressController API for subnet selection in AWS.
+- Support an install-time (Day 1) configuration for subnet selection of the default
+  IngressController.
+- Support upgrade compatibility and migration to the new API when using an unmanaged
+  `service.beta.kubernetes.io/aws-load-balancer-subnets` annotation.
+- Support the new API field for both Classic Load Balancers (CLB) and Network Load Balancers (NLB).
+
+### Non-Goals
+
+- Extend support to platforms other than AWS.
+- Automatically configure subnets for private clusters.
+- Extend the feature to ALBs or the AWS Load Balancer Operator (which already has API for subnet configuration).
+- Enable automatic updating of the load balancer service subnets by automatically deleting
+  the service, which is disruptive.
+
+## Proposal
+
+### IngressController API Proposal
+
+The `IngressController` API is extended by adding an optional parameter `Subnets`
+of type `[]string` to the `AWSLoadBalancerParameters` struct, to manage the
+`service.beta.kubernetes.io/aws-load-balancer-subnets` annotation on the
+LoadBalancer-type service.
+
+The `service.beta.kubernetes.io/aws-load-balancer-subnets` annotation is used
+for both CLBs and NLBs, hence adding it to `AWSLoadBalancerParameters` enables
+common configuration for both.
+
+While AWS, GCP, and Azure provide annotations for subnet configuration, AWS's
+annotation accepts multiple subnets, whereas GCP and Azure only permit a
+single subnet. For this reason, we made this API specific to AWS by adding
+the configuration in `AWSLoadBalancerParameters`.
+
+```go
+// AWSLoadBalancerParameters provides configuration settings that are
+// specific to AWS load balancers.
+// +union
+type AWSLoadBalancerParameters struct {
+  [...]
+
+  // subnets specifies the list of subnets for the load balancer to
+  // route traffic to. The values can be either a subnet ID or name.
+  // In order for the load balancer to be provisioned with subnets:
+  // * Each subnet must exist.
+  // * Each subnet must be from a different availability zone.
+  // * The load balancer service must be deleted.
+  //
+  // When omitted, the subnets will be auto-discovered per availability zone.
+  //
+  // +optional
+  // +openshift:enable:FeatureGate=IngressControllerLBSubnetsAWS
+  Subnets []string `json:"subnets,omitempty"`
```

Comment:
Should the value here be immutable? That would show users they can't change it without a delete and recreate (`--force`) and would make it very obvious that the load balancer will be deprovisioned and reprovisioned as part of this change

That may mean you can skip the 2 step approach with manual deletion of the service